assets/test/e2e/test.js

```diff
@@ -74,95 +165,136 @@ describe('', function() {
+        expect(element(by.cssContainingText(".component .service","database")).isPresent()).toBe(true);
+        expect(element(by.cssContainingText(".component .service","frontend")).isPresent()).toBe(true);
+        expect(element(by.cssContainingText(".component .route","www.example.com")).isPresent()).toBe(true);
+        expect(element(by.cssContainingText(".pod-template-build","Build: ruby-sample-build")).isPresent()).toBe(true);
+        expect(element(by.cssContainingText(".deployment-trigger","new image for origin-ruby-sample:latest")).isPresent()).toBe(true);
+        expect(element.all(by.css(".pod-running")).count()).toEqual(3);
+        // TODO: validate correlated images, builds, source
+      });
+
+      it('should browse builds', function() {
+        browser.get('/project/test/browse/builds');
+        expect(element(by.css('h1')).getText()).toEqual("Builds");
+        // TODO: validate presented strategies, images, repos
+      });
+
+      it('should browse deployments', function() {
+        browser.get('/project/test/browse/deployments');
+        expect(element(by.css('h1')).getText()).toEqual("Deployments");
+        // TODO: validate presented deployments
+      });
+
+      it('should browse events', function() {
+        browser.get('/project/test/browse/events');
+        expect(element(by.css('h1')).getText()).toEqual("Events");
+        // TODO: validate presented events
+      });
+
+      it('should browse image streams', function() {
+        browser.get('/project/test/browse/images');
+        expect(element(by.css('h1')).getText()).toEqual("Image Streams");
+        // TODO: validate presented images
+      });
+
+      it('should browse pods', function() {
+        browser.get('/project/test/browse/pods');
+        expect(element(by.css('h1')).getText()).toEqual("Pods");
+        // TODO: validate presented pods, containers, correlated images, builds, source
+      });
+
+      it('should browse services', function() {
+        browser.get('/project/test/browse/services');
+        expect(element(by.css('h1')).getText()).toEqual("Services");
+        // TODO: validate presented ports, routes, selectors
+      });
+
+      it('should browse settings', function() {
+        browser.get('/project/test/settings');
+        expect(element(by.css('h1')).getText()).toEqual("Project Settings");
+        // TODO: validate presented project info, quota and resource info
+      });
+
+      describe('when adding to project', function() {
+        it('should view the create page', function() { goToAddToProjectPage("test"); });
+
+        it('should create from source', function() {
+          var projectName = "test";
+          var sourceUrl = "https://github.com/openshift/rails-ex";
+          var appName = "my-rails-ex";
+          var builderImage = "ruby";
+          goToAddToProjectPage(projectName);
+          requestCreateFromSource(projectName, sourceUrl).then(function() {
```

Comment:
see above comment on promise usage. I'd sort of expect these to return success/failure, and do

if (!requestCreateFromSource(projectName, sourceUrl)) {
  return;
}
...